mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -93,8 +93,17 @@ internal static void Map (IntPtr handle, long offset, ref long size, MemoryMappe
 			}
 		}
 
+		static void CheckString (string name, string value)
+		{
+			// Native code tends to truncate at nul which is incorrect. Guard it here.
+			if (value != null && value.IndexOf((char)0) != -1)
+				throw new ArgumentException ($"{name}:{value} contains embedded nul.");
+		}
+
 		internal static IntPtr OpenFile (string path, FileMode mode, string mapName, out long capacity, MemoryMappedFileAccess access, MemoryMappedFileOptions options)
 		{
+			CheckString ("path", path);
+			CheckString ("mapName", mapName);
```

Comment:
these should use `nameof(path)` etc instead.

Reply:
nameof thank you. I didn't intend to change the handling of null. It likely already is handled non-silently. But if was going to be a null deref in native, we can either let it null deref in managed, or explicitly check, or let it keep null-derefing in native. I should write some tests, though I think the null deref will work ok at either level -- just that you can control the error message with an explicit check.